src/ngraph/descriptor/input.hpp

```diff
@@ -84,6 +94,8 @@ namespace ngraph
             Output* m_output;
 
         private:
+            bool m_is_relevant_to_shape;
```

Comment:
These might want to be sets of outputs (or indices) and we might want outputs to know sets of inputs they depend on. We'll see. Fine for now.